### PR TITLE
First implementation of the AccountModule

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

**Comment by hansl** [comment]
This PR includes #54 and #53.

**Review by fmorency — CHANGES_REQUESTED** [review]
Here's a little something before your vacations 🐈‍⬛ 

I did NOT review

- AccountMap*
- Tests

yet.

File: src/many/src/server/module/_9_account.rs
Comment (hansl) [reply src/many/src/server/module/_9_account.rs:210]: The spec is wrong here, this should definitely not be optional. Right?

File: src/many/src/server/module/_9_account.rs
Comment (hansl) [reply src/many/src/server/module/_9_account.rs:207]: This shouldn't be optional... There's always at least 1 role.

File: src/many/src/server/module/_9_account.rs
Comment (hansl) [reply src/many/src/server/module/_9_account.rs:114]: I'm gonna remove this from the spec. We don't really use it anywhere else.

File: src/many/src/server/module/_9_account.rs
Comment (fmorency) [inline src/many/src/server/module/_9_account.rs:322]: We should probably stick to a single nomenclature, either `name` or `description`.

File: src/many/src/server/module/_9_account.rs
Comment (fmorency) [inline src/many/src/server/module/_9_account.rs:428]: Add empty line

File: src/many/src/server/module/_9_account.rs
Comment (fmorency) [inline src/many/src/server/module/_9_account.rs:419]: Use `info()` when implemented instead of accessing the internals directly.

File: src/many/src/server/module/_9_account/features.rs
Comment (fmorency) [inline src/many/src/server/module/_9_account/features.rs:132]: You could `proctest` this thing and use arbitrary values

File: src/many/src/server/module/_9_account.rs
Comment (fmorency) [reply src/many/src/server/module/_9_account.rs:210]: Yes, `owner`.

File: src/many/src/server/module/_9_account.rs
Comment (fmorency) [reply src/many/src/server/module/_9_account.rs:213]: The `spec` says that we should at least pass 1 account feature when creating a new account. In that case, it should NOT be optional.

However, the `spec` also says `an account without features doesn't do anything and is an error.` and I'm not so sure about that. E.g.: we could want to create a new Account without any feature, having only the `owner` role. It doesn't do anything, but one could add new roles/features afterward using `addFeatures()`.

WDYT?

**Review by fmorency — COMMENTED** [review]
It's looking good; thanks a lot for this. The PR gives me some ideas for the future.

Some `todo!` still, but no major complaint.

Will you be finishing the mock implementation and related tests before merging?

Good work!

File: src/many/src/server/module/_9_account.rs
Comment (hansl) [reply src/many/src/server/module/_9_account.rs:303]: I actually also added this to the Account constructor.

File: src/many/src/server/module/_9_account.rs
Comment (hansl) [reply src/many/src/server/module/_9_account.rs:213]: I need to think about this a bit, but it doesn't prevent this PR, IMO.

**Comment by hansl** [comment]
The module tests are basically useless IMO. I need to rethink those, and I think the proper solution will be to have the module impl in many-rs and having a `Storage` trait that many-framework implements. So move some of the module logic from framework to many-rs. WDYT?

Anyway, this PR is ready to merge if it's all good for you.

**Comment by fmorency** [comment]
> The module tests are basically useless IMO. I need to rethink those, and I think the proper solution will be to have the module impl in many-rs and having a `Storage` trait that many-framework implements. So move some of the module logic from framework to many-rs. WDYT?
> 
> Anyway, this PR is ready to merge if it's all good for you.

Module testing is something I wanted to discuss with you at our next meeting.

I came to the same conclusion while implementing tests on my end; I needed to write a mock implementation for each module. While I think the uselessness of this strategy is debatable, we need to rethink our approach.

File: src/many/src/server/module/_9_account.rs
Comment (fmorency) [inline src/many/src/server/module/_9_account.rs:420]: nit: Remove entry from `account.roles` if value is empty after role removal?

**Review by fmorency — APPROVED** [review]
You can merge after addressing the comments. Thanks!

File: src/many/src/server/module/_9_account.rs
Comment (hansl) [reply src/many/src/server/module/_9_account.rs:420]: If we ever move logic to this crate we should, but this is test logic.

**Pull request merged**
